rust/src/dns/detect.rs

```diff
@@ -103,6 +121,357 @@ pub extern "C" fn SCDnsDetectRrtypeMatch(
     return 0;
 }
 
+static mut G_DNS_ANSWER_NAME_BUFFER_ID: c_int = 0;
+static mut G_DNS_QUERY_NAME_BUFFER_ID: c_int = 0;
+static mut G_DNS_QUERY_BUFFER_ID: c_int = 0;
+static mut G_DNS_OPCODE_KW_ID: c_int = 0;
+static mut G_DNS_OPCODE_BUFFER_ID: c_int = 0;
+static mut G_DNS_RCODE_KW_ID: c_int = 0;
+static mut G_DNS_RCODE_BUFFER_ID: c_int = 0;
+static mut G_DNS_RRTYPE_KW_ID: c_int = 0;
+static mut G_DNS_RRTYPE_BUFFER_ID: c_int = 0;
```

Comment:
This had me thinking... At some point I imagine, using the library to hold 2 completely distinct handles to different Suricata engines, with completely different configurations. These aren't the only globals that might be a problem.  Just something to keep in mind generally I think as we add globals.

Reply:
>  hold 2 completely distinct handles to different Suricata engines

big work indeed to remove globals...